core/src/com/unciv/ui/utils/ExtensionFunctions.kt

```diff
@@ -93,19 +93,48 @@ fun Actor.addBorder(size:Float, color: Color, expandCell:Boolean=false): Table {
     return table
 }
 
-fun Table.addSeparator(): Cell<Image> {
-    row()
-    val image = ImageGetter.getWhiteDot()
-    val cell = add(image).colspan(columns).height(2f).fill()
+/** get widget with background for a new separator */
+private fun getSeparatorTable(color: Color): Container<Actor> {
+    return Container<Actor>().apply {
+        background = when {
+            color.a != 0f -> ImageGetter.getBackground(color)
+            else -> CameraStageBaseScreen.skin.get("default-vertical", SplitPane.SplitPaneStyle::class.java).handle
+        }
+    }
+}
+
+/**
+ * Create a horizontal separator as an empty Container with a colored background.
+ * @receiver A Gdx [Table] to add this separator to
+ * @param color The color to use
+ * @param colSpan Number of columns to span (omit to use the current column count)
+ * @param height Height of the separator
+ * @return The Cell that has been added to the receiver Table
+ */
+fun Table.addSeparator(color: Color = Color.WHITE, colSpan: Int = 0, height: Float = 2f): Cell<Container<Actor>> {
+    if (!cells.isEmpty && !cells.last().isEndRow) row()
+    val separator = getSeparatorTable(color)
+    val cell = add(separator)
+        .colspan(if (colSpan == 0) columns else colSpan)
+        .minHeight(height).fillX()
     row()
     return cell
 }
 
-fun Table.addSeparatorVertical(): Cell<Image> {
-    val image = ImageGetter.getWhiteDot()
-    return add(image).width(2f).fillY()
+/**
+ * Create a vertical separator as an empty Container with a colored background.
+ * 
+ * Note: Unlike the horizontal [addSeparator] this cannot automatically span several rows. Repeat the separator if needed.
+ * @receiver A Gdx [Table] to add this separator to
+ * @param color The color to use
+ * @param width Width of the separator
+ * @return The Cell that has been added to the receiver Table
```

Comment:
I know I'm being a bit pedantic, but if it's either removing these now or being stuck with unneeded lines in these files for probably forever, then the first it is.

Reply:
> pedantic

Really? I'll need a big can of coffee before judging that... :judge: :hammer:

> now or forever

Precisely my motivation too - this is groundwork so others will build on it so it should be crystal clear to coders using it. IMHO the whole _file_ could do with more precise Kdoc's, I see myself revisiting it all the time to see how things tick or what other overloads might be there or... And those Studio tooltips could help with that (even if they're the main source of hard crashes). It tempts me again and again to add them to every single element in the file, and then I see the sheer number of them and get discouraged...

(Don't forget to expressly _say_ the final word on getImage vs getBackground)

Reply:
Yes - the function signature already offered by Studio in those tooltips is definitely enough here.